src/lightgbm/src/main/scala/LightGBMClassifier.scala

```diff
@@ -54,8 +54,10 @@ class LightGBMClassifier(override val uid: String)
      */
     val actualNumClasses = getNumClasses(dataset)
     val networkParams = NetworkParams(getDefaultListenPort, inetAddress, port)
+    val optionalWeightCol = if (hasWeightCol) Some(getWeightCol) else None
+
     val lightGBMBooster = df
-      .mapPartitions(TrainUtils.trainLightGBM(networkParams, getLabelCol, getFeaturesCol,
+      .mapPartitions(TrainUtils.trainLightGBM(networkParams, getLabelCol, getFeaturesCol, optionalWeightCol,
```

Comment:
Can probably put the get(weightCol) here if not used elsewhere

Reply:
done!